accounts/v1/client.go

```diff
@@ -16,14 +16,19 @@ import (
 // e.g. "https://accounts.sourcegraph.com".
 //
 // Users should prefer to use the top-level 'sams.NewAccountsV1' constructor instead.
```

Comment:
nit: update the docstring to mention giving a `nil` `baseTransport` means using the `http.DefaultTransport` given this is a public function after all (I did see the mention in later down in a different entry point)?